docs/design/dynamic-conf.md

```diff
@@ -0,0 +1,35 @@
+
+Volcano consists of three components: Scheduler, Webhook, and Controller. The Scheduler component schedules jobs by performing a series of actions and plugins to find the most suitable node for them. The Controller Manager manages the lifecycle of CRD (Custom Resource Definition) resources and is mainly composed of Queue ControllerManager, PodGroupControllerManager, and VCJob ControllerManager. The Admission component is responsible for validating CRD API resources.
+Currently, the startup configuration information for Volcano components is static and cannot be updated at runtime. For example, the log level of the Scheduler poses a significant burden on administrators during operation and maintenance. It is urgent to add the capability of dynamic and hot configuration updates to improve the efficiency of problem troubleshooting in production environments.
+
+### 1.2 dynamic configuration candidates
+
+The startup configuration information for Volcano components is currently static, and we need to select appropriate configurations and make them dynamic. I will identify all the static parameters used when launching Volcano and choose the configurations that are necessary to be dynamically adjustable under specific scenarios.
+
+```
+-v, --v: The level of verbosity for log levels.
+```
+
+#### Possible actual use scenarios
+
+1. `-v `The inability to update the log level of Volcano's Scheduler at runtime poses a significant burden on administrators during operation and maintenance. It is crucial to add the capability of dynamic configuration hot updates to enhance the efficiency of problem identification in production environments.
+
+### 1.3 Project scope
+
+Only focus on the static parameters that start the Schedule component
+
+## 2 Implementation Plan
+
+The vc-scheduler pod has an internal socket service to monitor changes in the klog.scok file. Because the socket service in the pod cannot be directly accessed in the k8s Node, the klog.sock file needs to be mounted from the pod to the cluster. Use The curl command modifies the klog.sock file.
+[![](https://camo.githubusercontent.com/a5a8fabc657ba344e04992e0844d634dca711dae88954c0a80df6edf38eeeb3e/68747470733a2f2f702e697069632e7669702f3178343533372e706e67#from=url&id=gptPv&originHeight=720&originWidth=1074&originalType=binary&ratio=2&rotation=0&showTitle=false&status=done&style=none&title=)](https://camo.githubusercontent.com/a5a8fabc657ba344e04992e0844d634dca711dae88954c0a80df6edf38eeeb3e/68747470733a2f2f702e697069632e7669702f3178343533372e706e67)
+example curl command example
+
+```
+sudo curl --unix-socket /tmp/socks/klog.sock "http://localhost/setlevel?level=5&duration=60s"
+```
+
+## 3、Why not change klog level throuth change configmap
+Volcano is a commercial project. Volcano is deployed in a customer cluster. The configmap cannot be modified easily, but curl can be executed. I thought of using curl to pass in the klog value that you want to modify through the scoket method.
```

Comment:
I don't understand here, I think modifying the log level and modifying the configmap are both things done by the platform administrator, using curl is more intrusive. And I don't think this approach is very applicable, for example if my kubelet is running in a container, will it still work?

Reply:
Can you describe in detail the scenario where kubelet runs in a container?
Other reasons to use the curl scheme
1、Modifying configmap may affect other plugins
2、It'll be confusing that both options and configmap has same parameters.

Reply:
To be honest, I'm not sure about the benefits of running the kubelet in a container, but I do see a lot of cases at work where the kubelet is deployed in a container, and I'm guessing it might be for ease of deployment. rancher also runs the kubelet in a container.
When a kubelet is running in a container, the hostpath mounts the directory in the kubelet container, not the directory on the real host

Reply:
This affects only the volcano scheduler container. The mounting path is mapped from the volcano scheduler container to the host where the container is located. The kubelet in the container is not affected.

Reply:
> To be honest, I'm not sure about the benefits of running the kubelet in a container, but I do see a lot of cases at work where the kubelet is deployed in a container, and I'm guessing it might be for ease of deployment. rancher also runs the kubelet in a container. When a kubelet is running in a container, the hostpath mounts the directory in the kubelet container, not the directory on the real host

you mean the k8s deployed in container，so the kubelet deployed in the container？

Reply:
> you mean the k8s deployed in container，so the kubelet deployed in the container？

no, just the kubelet in container.

```
docker inspect kubelet
```

Reply:
for curl it, I think it will be OK, because the kubelet container is hostnetwork, but I feel this way (by curl) is not 'k8s', I can't think of any component that uses this method.

Reply:
@hwdef 
The usage scenario is that the log level needs to be temporarily raised in the customer's cluster to facilitate locating the cause of the error. However, changing the user configuration requires an apply, which is not required when using the curl command. Do you think there is any way in k8s to achieve the ability to change the log level without changing the configuration?

Reply:
Why refuse `apply`?

Reply:
The original solution was to modify the configmap, which contains the plugin configuration, and modifying the configmap may affect the plugin due to errors in modifying the configuration，There are risks.

Reply:
The log level is configured in the startup parameters and does not require modification of the configmap.

https://github.com/volcano-sh/volcano/blob/0d6d3b6a0dee7fffbc4e71e41f758d6d206e92bb/installer/volcano-development.yaml#L9178

Reply:
If you do this, the program will restart, and programs running online should not restart at will.

Reply:
volcano just a scheduler program. Restarting it won't give you a serious error.
Even in normal use, it can restart for a variety of reasons, so volcano-scheduler is deployed using deployment
If you are concerned about restart , then you should scale volcano-scheduler horizontally, e.g. to 3 replicas, so that even if one replica restart, the other two replicas are still working